pylabrobot/resources/resource_holder.py

```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+from pylabrobot.resources.coordinate import Coordinate
+from pylabrobot.resources.resource import Resource
+
+def get_child_location(resource: Resource) -> Coordinate:
+  """
+  If a resource is rotated, its rotated around its local origin. This does not always
+  match up with the parent resource's origin. This function calculates the difference
+  between the two origins and calculates the location of the resource correctly.
+  """
+  if not resource.rotation.y == resource.rotation.x == 0:
+    raise ValueError("Resource rotation must be 0 around the x and y axes")
+  if not resource.rotation.z % 90 == 0:
+    raise ValueError("Resource rotation must be a multiple of 90 degrees on the z axis")
+  location = {
+    0.0: Coordinate(x=0, y=0, z=0),
+    90.0: Coordinate(x=resource.get_size_y(), y=0, z=0),
+    180.0: Coordinate(x=resource.get_size_y(), y=resource.get_size_x(), z=0),
+    270.0: Coordinate(x=0, y=resource.get_size_x(), z=0),
+  }[resource.rotation.z % 360]
+  return location
+
+class ResourceHolderMixin:
```

Comment:
using a mixin avoids complexity arising from "diamond inheritance"

Reply:
how do we make sure `ResourceHolderMixin.assign_child_resource` has the same method header as `Resource`?

Reply:
i'm not sure i follow. isnt this the same as with a normal subclassing?

Reply:
bumping this

Reply:
any more thoughts on this @rickwierenga?